Sources/Testing/Testing.docc/testing-for-errors-in-swift-code.md

```diff
@@ -0,0 +1,100 @@
+# Testing for errors in Swift code
+
+<!--
+This source file is part of the Swift.org open source project
+
+Copyright (c) 2024 Apple Inc. and the Swift project authors
+Licensed under Apache License v2.0 with Runtime Library Exception
+
+See https://swift.org/LICENSE.txt for license information
+See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+Ensure that your code handles errors in the way you expect.
+
+## Overview
+
+The Swift language provides an idiomatic approach to [error handling](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/errorhandling), based on throwing errors where your code detects a failure for a caller to catch and react to.
+
+Write tests for your code that validate the conditions in which the code throws errors, and the conditions in which it returns without throwing an error.
+Use overrides of the `#expect()` and `#require()` macros that check for errors.
+
+### Validate that your code throws an expected error
+
+The Swift structure in this example represents a list that accepts any number of "tags" for items in the list.
```

Comment:
"Tags" are a term of art in swift-testing. To avoid confusion, use a different concept in this example.

Reply:
_checks synonyms_